13-Uncertainty/astropy_uncertainty.ipynb

```diff
@@ -0,0 +1,949 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "piscA_mr.pdf_median()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The above compute the mean, standard deviation, and median of the *samples* from the distribution. It is apparent that this reproduces the input we gave but only to ~1%, as expected from the number of samples.  We can also access the samples directly, as needed to produce, for example, a plot of the distribution:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.hist(piscA_mr.distribution, bins='auto', density=True, histtype='step')\n",
+    "\n",
+    "plt.axvline(piscA_mr.pdf_mean(), color='k')\n",
+    "plt.axvline(piscA_mr.pdf_mean() + piscA_mr.pdf_std(), color='k', ls=':')\n",
+    "plt.axvline(piscA_mr.pdf_mean() - piscA_mr.pdf_std(), color='k', ls=':');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we can represent both galaxies from this paper the same way,  letting us do quick convenient operations over them:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "piscB_mr = uncertainty.normal(17.18*u.mag, std=0.07*u.mag, n_samples=10000)\n",
+    "\n",
+    "for distr in [piscA_mr, piscB_mr]:\n",
+    "    plt.hist(distr.distribution, bins='auto', density=True, histtype='step')\n",
+    "    \n",
+    "    plt.axvline(distr.pdf_mean(), color='k')\n",
+    "    plt.axvline(distr.pdf_mean() + distr.pdf_std(), color='k', ls=':')\n",
+    "    plt.axvline(distr.pdf_mean() - distr.pdf_std(), color='k', ls=':')\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This on its own is enough to ask simple statistical questions using just the samples.  For example: what is the probability that Pisces A is brighter than Pisces B?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gridA, gridB = np.meshgrid(piscA_mr.distribution, piscB_mr.distribution)\n",
+    "np.sum(gridA < gridB) / gridA.size"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Depending on your computer's speed, you may have seen a noticeable amount of time passing for that computation. This is because the `meshgrid` function created an array element for every possible pair in the two distributions, which is $10000^2 = 10^8$, meaning it had to create two 100 million element array.  This probably used a few GB of your computer's memory, a not-insignificant fraction of what you have. This is one of the \"gotchas\" to be aware of with the monte carlo method: when doing operations that involve multiple distributions you often need to compare all combinations, which quickly becomes exponentially large and overwhelms your computer.\n",
```

Comment:
```suggestion
    "Depending on your computer's speed, you may have seen a noticeable amount of time passing for that computation. This is because the `meshgrid` function created an array element for every possible pair in the two distributions, which is $10000^2 = 10^8$, meaning it had to create two 100-million element arrays.  This likely a significant amount of your computer's memory. This is one of the \"gotchas\" to be aware of with the Monte Carlo method used in `uncertainty`: when doing operations that involve multiple distributions you often need to compare all combinations, which quickly becomes expensive and overwhelms your computer.\n",
```